app/src/main/java/clarkson/ee408/tictactoev4/LoginActivity.java

```diff
@@ -40,31 +49,56 @@ public void handleLogin() {
         String username = usernameField.getText().toString();
         String password = passwordField.getText().toString();
 
-        // TODO: verify that all fields are not empty before proceeding. Toast with the error message
+        // Verify that all fields are not empty before proceeding. Toast with the error message
+        if (username.isEmpty() || password.isEmpty()) {
+            Toast.makeText(this, "Please enter both username and password", Toast.LENGTH_SHORT).show();
+            return;
+        }
 
-        // TODO: Create User object with username and password and call submitLogin()
+        // Create User object with username and password and call submitLogin()
+        User user = new User(username, password);
+        submitLogin(user);
     }
-
     /**
      * Sends a LOGIN request to the server
      * @param user User object to login
      */
     public void submitLogin(User user) {
-        // TODO: Send a LOGIN request, If SUCCESS response, call gotoPairing(), else, Toast the error message from sever
+        // Send a LOGIN request, If SUCCESS response, call gotoPairing(), else, Toast the error message from sever
+        String jsonRequest = gson.toJson(user);
```

Comment:
Need to build the Request object for the sendRequest method of the SocketClient

```suggestion
        String serializedUser = gson.toJson(user);
        Request request = new Request(RequestType.LOGIN, serializedUser);
```